### ISSUE TITLE
Do bots think they don't have ammo?

### ISSUE THREAD

**Issue by Gvbavx** [wootguy/rcbot#5]
I'm confused, do the bots think they have no ammo? Once their weapon magazine is empty, bots will not reload (sometimes they will), ending up switching crowbar, they still keep using crowbars after respawning or getting a new weapon. Eventually All bots use crowbar
Picture:![20231022142136_1](https://github.com/wootguy/rcbot/assets/145967852/5f96a112-3118-482a-9b11-5f06b2c1cfa9)
Videos:
https://github.com/wootguy/rcbot/assets/145967852/764e49f0-e5e8-4ffe-940a-ad3fc7a0f3f0
https://github.com/wootguy/rcbot/assets/145967852/3a334c13-f76d-48c7-b423-08591198adbc
(Sorry for the bad quality)

Also I gave the bot everything, it'll try using Gluon gun, grenade, grapple

https://github.com/wootguy/rcbot/assets/145967852/161d9422-7481-48e1-bb24-28095780b7ce
https://github.com/wootguy/rcbot/assets/145967852/2e78f0a9-d098-4927-878d-929a38219035

Another thing I found that the bot was trying to destroy the breakables with grapple?
https://github.com/wootguy/rcbot/assets/145967852/403d328a-14a5-4ead-9130-276b1944382e

If the bots don't have a crowbar when respawn(in some maps you won't get a crowbar when spawn), they can use still other weapons(sometimes they keep switching weapons??), but if they get crowbar, 
the same thing happened

When a bot doesn't have a crowbar:
https://github.com/wootguy/rcbot/assets/145967852/3dba8c7f-99d3-413d-951a-8ac69c11b433

When a bot have a crowbar:
https://github.com/wootguy/rcbot/assets/145967852/bb13dd95-0ef0-47df-ae56-0c1978e07e1d

The same thing happened when a bot only has a pipe wrench:
https://github.com/wootguy/rcbot/assets/145967852/6c1e8bca-81e6-47cb-8cb8-088108066b37

In conclusion, Once a bot's weapon magazine is empty, bot will not reload (sometimes they will), ending up switching melee weapon

**Comment by Gvbavx** [comment]
The bots only have melee weapon when spawn, and get a new weapon:https://github.com/wootguy/rcbot/assets/145967852/07988051-2218-4d6b-915a-206ad4a97cab
Almost all bots are using crowbar

**Comment by Gvbavx** [comment]
It's odd that the bots won't use melee weapon in this map?
![20231022174454_1](https://github.com/wootguy/rcbot/assets/145967852/0a0e852e-0e8b-4953-811d-3fed0d2f8368)

**Comment by wootguy** [comment]
I thought I saw this when first testing the bots, but it hasn't happened since then. Now I see them start out holding the crowbar then switching to a ranged weapon after seeing an enemy. They know how much ammo they have but are wrong about the clip. This .dll will fix that and add a command `rcbot debug wep 1` to tell you why they don't use certain weapons (will flood your console).

[rcbot_wep_debug.zip](https://github.com/wootguy/rcbot/files/13064215/rcbot_wep_debug.zip)

![image](https://github.com/wootguy/rcbot/assets/12087544/3fb8af2f-fafb-4499-b471-a04d0b98e98f)

I'll make a separate issue for bots using the wrong weapon for breakables.

**Comment by Gvbavx** [comment]
I appreciate your help, it was really annoying when the bots were using crowbars XD I'll test it when I have some time :D

**Comment by Gvbavx** [comment]
The release zip doesn't has debug weapon info?
![image](https://github.com/wootguy/rcbot/assets/145967852/871c5490-8090-4200-a2fd-07e6d3038636)

**Comment by wootguy** [comment]
hmm try now, smth went wrong when I replaced the .zip

**Comment by Gvbavx** [comment]
Not sure why this still happened when I added too many bots, I was sure they have ammo and they didn't reload
I couldn't use rcbot debug wep 1 because It made my server lagging
![20231029185818_1](https://github.com/wootguy/rcbot/assets/145967852/c54f091a-503d-4596-95c6-395832c9ee75)
![20231029185821_1](https://github.com/wootguy/rcbot/assets/145967852/5a641d15-c95e-4dc1-8ba8-bc250c3db57a)
